### PR TITLE
Undo temporary policy that disallows changes to collection.xml

### PR DESCRIPTION
**Pull request by karenc**
The temporary policy was added because of a corner case that prevented
us from making Press only publish changed content.  Since there will be
a fix for this, we can remove the temporary policy.

----

Part 1/3 of #190

### CODE REVIEW

File: tests/functional/views/test_legacy_publishing.py [hunk 1]
```diff
@@ -304,16 +299,18 @@ def test_publishing_overwrite_module_litezip(
...
+#    assert resp.status_code == 400
+#    expected_msgs = [
+#        {
+#            "id": 4,
+#            "message": "collection changed",
+#            "item": collection.id,
+#            "error": 'modifying a collection is temporarily disallowed'
+#        }
+#    ]
+#    assert resp.json['messages'] == expected_msgs
```
Comment (pumazi) [inline tests/functional/views/test_legacy_publishing.py]: Missed taking this out. TODO

File: tests/functional/views/test_legacy_publishing.py [hunk 2]
```diff
@@ -365,7 +362,7 @@ def test_publishing_overwrite_collection_litezip(
     )
     # FIXME: uncomment.
     # assert resp.status_code == 204
```
Comment (pumazi) [inline tests/functional/views/test_legacy_publishing.py]: Remove the `FIXME` too. TODO

File: tests/functional/views/test_legacy_publishing.py [hunk 3]
```diff
@@ -365,7 +362,7 @@ def test_publishing_overwrite_collection_litezip(
...
-    assert resp.status_code == 400  # no changes to collection.xml
+    assert resp.status_code == 200
```
Comment (pumazi) [inline tests/functional/views/test_legacy_publishing.py]: Hmm... was this previously 204 and it is now 200?

Comment (karenc) [reply tests/functional/views/test_legacy_publishing.py]: Yeah.. I'm not sure what this means

**Comment by karenc** [comment]
I updated the changelog because 7.3.0 has been released and so I don't need to group the changes into 8.0.0 anymore.

Also changed the commit author to be Bryan.

File: tests/functional/views/test_legacy_publishing.py (hunk 1 above)
Comment (pumazi) [reply tests/functional/views/test_legacy_publishing.py]: I've been reassured out-of-band that this will come back into a functioning state in a followup PR
